### PR TITLE
Additional docs for Redfish Exporter

### PR DESCRIPTION
**Pull request by AMD-melliott**
This PR provides an updated `README.md`, and adds these additional docs:
- `docs\configuration.md` - Configuration Guide, environment variables,  Docker usage
- `docs\metrics.md` - List of metrics exposed
- `docs\testing.md` - Mock server usage, integration testing
- `docs\user-guide.md` - Dedicated user guide. Currently there is a lot of overlap with `README.md`, so this doc may be unnecessary

I've run through these commands on my own but would appreciate if the team can double check everything for accuracy and provide feedback on anything missing/needed edits.

### CODE REVIEW

**Comment by AMD-melliott** [comment]
Resolved merge conflicts with README.md

File: redfish-exporter/README.md [hunk 1]
```diff
@@ -1,158 +1,87 @@
...
-    ]
- }
-```
+### Prerequisites
 
-Send a POST request to register this subscription:
-```bash
-curl -d "@data.json" -X POST http://localhost:8000/redfish/v1/EventService/Subscriptions
-```
+- Go 1.23 or later
```
Comment (araslanix) [inline redfish-exporter/README.md:17]: Docker?

File: redfish-exporter/docs/metrics.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+## Event Metrics
+
+### `RedFishEvents_recieved`
+
+- **Type**: Counter
+- **Description**: Total number of Redfish events received
+- **Labels**:
+  - `SourceIP`: IP address of the Redfish server that sent the event
+  - `EventType`: Type of the Redfish event
```
Comment (araslanix) [inline redfish-exporter/docs/metrics.md:13]:
```suggestion
  - `Severity`: Severity of the Redfish event
```
New requirements came in so I've updated those :)

File: redfish-exporter/docs/metrics.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Metrics Documentation
+
+The AMD Redfish Exporter exposes the following Prometheus-compatible metrics:
+
+## Event Metrics
+
+### `RedFishEvents_recieved`
```
Comment (araslanix) [inline redfish-exporter/docs/metrics.md:7]:
```suggestion
### `RedFishEvents_received`
```
A typo that's now fixed.

**Review by araslanix — APPROVED** [review]
Looks great!

File: redfish-exporter/README.md (hunk 1 above)
Comment (araslanix) [reply redfish-exporter/README.md:17]: The reason is that the make command will run a Docker in a this step.
https://github.com/nod-ai/ADA/blob/70bfacd599fa440b52750b00aa2184b84621eb1b/redfish-exporter/api/Makefile#L4

Comment (AMD-melliott) [reply redfish-exporter/README.md:17]: Good call. Added Docker to the prerequisites in the readme and user guide

File: redfish-exporter/docs/metrics.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  - `Severity`: Severity of the Redfish event
+
+### `RedFishEvents_processing_time`
+
+- **Type**: Gauge
+- **Description**: Time taken to process Redfish events in seconds
+- **Labels**:
+  - `SourceIP`: IP address of the Redfish server that sent the event
+  - `Severity`: Severity of the Redfish event
+
```
Comment (yuva29) [inline redfish-exporter/docs/metrics.md:22]: Do we need add `SlurmAPI_failure` and `SlurmAPI_success` metrics that's defined here https://github.com/nod-ai/ADA/blob/main/redfish-exporter/metrics/metrics.go#L39-L53?

File: redfish-exporter/docs/configuration.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+- `LISTENER_IP`: IP address to listen on (default: "127.0.0.1")
+- `LISTENER_PORT`: Port to listen on for Redfish events (default: "8080")  (API events?)
+- `USE_SSL`: Enable SSL/TLS (default: "false")
+- `CERTFILE`: Path to SSL certificate file
+- `KEYFILE`: Path to SSL key file
+- `METRICS_PORT`: Port to expose Prometheus metrics (default: "2112")
+- `SLURM_TOKEN`: SLURM authentication token
+- `SLURM_CONTROL_NODE`: SLURM control node address
+- `SUBSCRIPTION_PAYLOAD`: JSON string containing Redfish subscription details
+- `REDFISH_SERVERS`: JSON array of Redfish server configurations
```
Comment (yuva29) [inline redfish-exporter/docs/configuration.md:18]: Under `REDFISH_SERVERS`, they need to populate the slurm node name correctly for the drain to happen:

`REDFISH_SERVERS=[{"ip": "http://<ip>:<port>", "username": "<username>", "password": "<password>", "loginType": "Session", "slurmNode": "Node1"}]`

File: redfish-exporter/README.md [hunk 6]
```diff
@@ -176,15 +106,25 @@ Note that SSL can be enabled using the provided dev certificate and key and `--s
     ./amd-redfish-exporter --enable-slurm
     ```
 
-### Integration Testing
+- The exporter will start subscribing to events from the configured Redfish servers and expose metrics on the specified metrics port.
```
Comment (yuva29) [inline redfish-exporter/README.md:109]: They are listed under slurm integration. Could you please check if it's the correct placement?

File: redfish-exporter/docs/metrics.md (hunk 4 above)
Comment (AMD-melliott) [reply redfish-exporter/docs/metrics.md:22]: Added these to `metrics.md`

File: redfish-exporter/docs/configuration.md (hunk 5 above)
Comment (AMD-melliott) [reply redfish-exporter/docs/configuration.md:18]: Added a note to `configuration.md` to call this out

File: redfish-exporter/README.md (hunk 6 above)
Comment (AMD-melliott) [reply redfish-exporter/README.md]: I think there were some formatting issues introduced when I resolved a merge conflict. This is sorted out.

**Comment by AMD-melliott** [comment]
Thank you for the feedback @yuva29, I've incorporated your suggestions into the docs.

File: redfish-exporter/docs/metrics.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+### `SlurmAPI_success`
+
+- **Type**: Gauge
+- **Description**: Total number of Slurm API calls that succeeded
+- **Labels**:
+  - `SourceIP`: IP address of the SLURM node
```
Comment (yuva29) [inline redfish-exporter/docs/metrics.md:28]: This is the IP address of redfish server.

**Review by yuva29 — APPROVED** [review]
One minor comment and everything else is great. Thanks.

File: redfish-exporter/docs/metrics.md (hunk 7 above)
Comment (AMD-melliott) [reply redfish-exporter/docs/metrics.md]: Gotcha. Fixed!

**Pull request merged**